superset/assets/javascripts/explorev2/components/ChartContainer.jsx

```diff
@@ -149,8 +158,14 @@ class ChartContainer extends React.Component {
     this.props.actions.removeChartAlert();
   }
 
-  renderVis() {
-    visMap[this.props.viz_type](this.state.mockSlice).render();
+  renderVisOnChange(heightChange) {
+    if (this.props.chartStatus === 'loading') {
+      this.state.viz.render();
+    }
+    if (heightChange) {
+      this.state.viz.render();
+      this.state.viz.resize();
```

Comment:
do we split these up so resize can be called separately? seems to me that the render call should take into account the new container size and render accordingly.